packages/nuxt/src/runtime/plugins/route-detector.server.ts

```diff
@@ -0,0 +1,50 @@
+import { getActiveSpan, getRootSpan, logger, SEMANTIC_ATTRIBUTE_SENTRY_SOURCE } from '@sentry/core';
+import { defineNuxtPlugin } from 'nuxt/app';
+import type { NuxtPage } from 'nuxt/schema';
+import { extractParametrizedRouteFromContext } from '../utils/route-extraction';
+
+export default defineNuxtPlugin(nuxtApp => {
+  nuxtApp.hooks.hook('app:rendered', async renderContext => {
+    let buildTimePagesData: NuxtPage[] = [];
+    try {
+      // This is a common Nuxt pattern to import build-time generated data: https://nuxt.com/docs/4.x/api/kit/templates#creating-a-virtual-file-for-runtime-plugin
+      // @ts-expect-error This import is dynamically resolved at build time (`addTemplate` in module.ts)
+      const { default: importedPagesData } = await import('#build/sentry--nuxt-pages-data.mjs');
+      buildTimePagesData = importedPagesData || [];
+    } catch (error) {
+      buildTimePagesData = [];
+    }
```

Comment:
l: should we log the error? Otherwise

```suggestion
    } catch {
      buildTimePagesData = [];
    }
```